src/Providers/Http/DTO/Response.php

```diff
@@ -0,0 +1,267 @@
+
+use InvalidArgumentException;
+use WordPress\AiClient\Common\AbstractDataTransferObject;
+
+/**
+ * Represents an HTTP response.
+ *
+ * This class encapsulates HTTP response data that has been converted
+ * from PSR-7 responses by the HTTP transporter.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-type ResponseArrayShape array{
+ *     statusCode: int,
+ *     headers: array<string, string|list<string>>,
+ *     body?: string|null,
+ *     reasonPhrase: string
+ * }
+ *
+ * @extends AbstractDataTransferObject<ResponseArrayShape>
+ */
+class Response extends AbstractDataTransferObject
+{
+    public const KEY_STATUS_CODE = 'statusCode';
+    public const KEY_HEADERS = 'headers';
+    public const KEY_BODY = 'body';
+    public const KEY_REASON_PHRASE = 'reasonPhrase';
+
+    /**
+     * @var int The HTTP status code.
+     */
+    protected int $statusCode;
+
+    /**
+     * @var array<string, string|list<string>> The response headers.
+     */
+    protected array $headers;
+
+    /**
+     * @var string|null The response body.
+     */
+    protected ?string $body;
+
+    /**
+     * @var string The reason phrase.
+     */
+    protected string $reasonPhrase;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param int $statusCode The HTTP status code.
+     * @param array<string, string|list<string>> $headers The response headers.
+     * @param string|null $body The response body.
+     * @param string $reasonPhrase The reason phrase.
+     *
+     * @throws InvalidArgumentException If the status code is invalid.
+     */
+    public function __construct(int $statusCode, array $headers, ?string $body = null, string $reasonPhrase = '')
+    {
+        if ($statusCode < 100 || $statusCode >= 600) {
+            throw new InvalidArgumentException('Invalid HTTP status code: ' . $statusCode);
+        }
+
+        $this->statusCode = $statusCode;
+        $this->headers = $headers;
+        $this->body = $body;
+        $this->reasonPhrase = $reasonPhrase;
+    }
+
+    /**
+     * Gets the HTTP status code.
+     *
+     * @since n.e.x.t
+     *
+     * @return int The status code.
+     */
+    public function getStatusCode(): int
+    {
+        return $this->statusCode;
+    }
+
+    /**
+     * Gets the response headers.
+     *
+     * @since n.e.x.t
+     *
+     * @return array<string, string|list<string>> The headers.
+     */
+    public function getHeaders(): array
+    {
+        return $this->headers;
+    }
+
+    /**
+     * Gets a specific header value.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $name The header name (case-insensitive).
+     * @return string|list<string>|null The header value(s) or null if not found.
+     */
+    public function getHeader(string $name)
+    {
+        // Case-insensitive header lookup
+        foreach ($this->headers as $key => $value) {
+            if (strcasecmp($key, $name) === 0) {
+                return $value;
+            }
+        }
```

Comment:
Instead of this `foreach` loop, how about we add a private lookup map property where each key is transformed to lower-case? This way here you could just check for `isset($this->lookup_map[strtolower($name)])`, no loop needed.

Reply:
Resolved in 75c69c1